src/cephadm/cephadm

```diff
@@ -2404,157 +2404,162 @@ def command_adopt():
-                os.rename(simple_fn, new_fn)
-                logger.info('Disabling host unit ceph-volume@ simple unit...')
-                call_throws(['systemctl', 'disable',
-                             'ceph-volume@simple-%s-%s.service' % (
-                                 daemon_id, osd_fsid)])
-            else:
-                # assume this is an 'lvm' c-v for now, but don't error
-                # out if it's not.
-                logger.info('Disabling host unit ceph-volume@ lvm unit...')
-                call(['systemctl', 'disable',
-                      'ceph-volume@lvm-%s-%s.service' % (daemon_id, osd_fsid)])
-
-        # config
-        config_src = '/etc/ceph/%s.conf' % (args.cluster)
-        config_src = os.path.abspath(args.legacy_dir + config_src)
-        config_dst = os.path.join(data_dir_dst, 'config')
-        copy_files([config_src], config_dst, uid=uid, gid=gid)
-
-        # logs
-        logger.info('Moving logs...')
-        log_dir_src = ('/var/log/ceph/%s-%s.%s.log*' %
-                        (args.cluster, daemon_type, daemon_id))
-        log_dir_src = os.path.abspath(args.legacy_dir + log_dir_src)
-        log_dir_dst = make_log_dir(fsid, uid=uid, gid=gid)
-        move_files(glob(log_dir_src),
-                   log_dir_dst,
-                   uid=uid, gid=gid)
-
-        logger.info('Creating new units...')
-        make_var_run(fsid, uid, gid)
-        c = get_container(fsid, daemon_type, daemon_id)
-        deploy_daemon_units(fsid, uid, gid, daemon_type, daemon_id, c,
-                            enable=True,  # unconditionally enable the new unit
-                            start=(state == 'running'),
-                            osd_fsid=osd_fsid)
-        update_firewalld(daemon_type)
-
-    else:
-        raise Error('adoption of style %s not implemented' % args.style)
-
-def command_adopt_prometheus(daemon_id, uid, gid):
-    # type: (str, int, int) -> None
-
-    daemon_type = 'prometheus'
+    # legacy check
     if args.style != 'legacy':
         raise Error('adoption of style %s not implemented' % args.style)
 
+    # lock
     fsid = get_legacy_daemon_fsid(args.cluster,
                                   daemon_type,
                                   daemon_id,
                                   legacy_dir=args.legacy_dir)
     if not fsid:
         raise Error('could not detect legacy fsid; set fsid in ceph.conf')
-
     l = FileLock(fsid)
     l.acquire()
 
+    # call correct adoption
+    if daemon_type in Ceph.daemons:
+        command_adopt_ceph(daemon_type, daemon_id, fsid);
+    elif daemon_type == 'prometheus':
+        command_adopt_prometheus(daemon_id, fsid)
+    elif daemon_type == 'grafana':
+        raise Error('adoption of grafana not implemented')
+    elif daemon_type == 'node-exporter':
+        raise Error('adoption of node-exporter not implemented')
+    elif daemon_type == 'alertmanager':
+        raise Error('adoption of alert-manager not implemented')
```

Comment:
seems like this is the same as the default `else` block?

Reply:
Figured I would be explicit about what's next to be implemented.... although I could see this collapsing into a `Monitoring.components` check